lib/batch/batch_metadata_updates_655.rb

```diff
@@ -0,0 +1,27 @@
+# Transfer metadata values from Original Publisher to Digital Publisher field
+
+# below is too slow
+#items_needing_transfer = GenericFile.all.reject { |gf| 
+#                           gf.original_publisher == []
+#                         }
+
+solr_items_w_original_publisher_value = ActiveFedora::SolrService.query(
+                                     'original_publisher_tesim:*', 
+                                     { rows: 99999 }
+                                   )
+
+solr_items_w_original_publisher_value.each do |solr_item|
+  # could be either Collection or GenericFile, so we convert the string to
+  # class to get the object by id
+  constant_item = solr_item['has_model_ssim'].first.constantize
+  item = constant_item.find(solr_item['id'])
+
+  item.original_publisher.each do |op|
+    if !item.publisher.include?(op)
+      item.publisher += [op]
+    end
+  end
+
+  item.original_publisher -= item.original_publisher
+  item.save!
```

Comment:
Does this `save!` reindex? If entries are in DB *and* index we want to make sure both are in sync.

Reply:
Yes, `save!` reindexes, and keeps Fedora and index objects in sync.